### PR TITLE
Allow offline VDI moving. Fix vatesfr/xo-web/issue…

### PR DESCRIPTION
**Pull request by olivierlambert**
…s/591

### CODE REVIEW

**Comment by julien-f** [comment]
Commit message should be `Allow offline VDI moving (fix vatesfr/xo-web#591)`.

**Comment by julien-f** [comment]
My bad, it was just the title of the PR.

File: src/xapi.js [hunk 1]
```diff
@@ -1385,6 +1385,33 @@ export default class Xapi extends XapiBase {
...
+      if (error.code !== 'VDI_NEEDS_VM_FOR_MIGRATE') {
+        throw error
+      }
+      const newVdiref = await this.call('VDI.copy', vdi.$ref, sr.$ref)
+      const newVdi = await this._getOrWaitObject(newVdiref)
+      await Promise.all(mapToArray(vdi.$VBDs, async vbd => {
+        // Remove the old VBD
+        await this.call('VBD.destroy', vbd.$ref)
+        // Attach the new VDI to the VM with old VBD settings
+        await this._createVbd(vbd.$VM, newVdi, {
```
Comment (julien-f) [inline src/xapi.js]: Would'nt it be possible to attach the new VBD before destroying the old one?

File: src/xapi.js [hunk 2]
```diff
@@ -1385,6 +1385,33 @@ export default class Xapi extends XapiBase {
...
+      await this.call('VDI.pool_migrate', vdi.$ref, sr.$ref, {})
+    } catch (error) {
+      if (error.code !== 'VDI_NEEDS_VM_FOR_MIGRATE') {
+        throw error
+      }
+      const newVdiref = await this.call('VDI.copy', vdi.$ref, sr.$ref)
+      const newVdi = await this._getOrWaitObject(newVdiref)
+      await Promise.all(mapToArray(vdi.$VBDs, async vbd => {
+        // Remove the old VBD
+        await this.call('VBD.destroy', vbd.$ref)
```
Comment (julien-f) [inline src/xapi.js]: Is it necessary to destroy the VDI or is automatically garbage-collected when destroying the VDI?

File: src/xapi.js (hunk 1 above)
Comment (olivierlambert) [inline src/xapi.js]: Nope: what about the position? It's already occupied by the old one.

File: src/xapi.js (hunk 2 above)
Comment (olivierlambert) [inline src/xapi.js]: AFAIK, it's only VBDs which are GC.

Comment (julien-f) [inline src/xapi.js]: I meant destroying the VBD -_-"

File: src/xapi.js (hunk 1 above)
Comment (julien-f) [inline src/xapi.js]: Fair enough.

**Pull request merged**